### PR TITLE
New example

### PR DESCRIPTION
**Pull request by elalish**
We've been talking about improving our docs, and users seem to be finding the @mrdoob tester https://model-viewer-tester.glitch.me/ to be quite useful, both for testing out model-viewer and for reporting bugs. Therefore, I've build a docs page that mimics it. It'll need some work yet and it's not intended for 0.8, but I wanted to float the idea and get some feedback.

### CODE REVIEW

**Comment by elalish** [comment]
@cdata This is ready for review. One question: how should we handle attribution? I need to add quite a few more; should they be behind a link or something?

**Comment by elalish** [comment]
@cdata I started to add the TS version, but it's giving me a lot of errors when I build that don't show up in VSCode. Also, it really doesn't like the for loop over properties; is there an efficient way to do this in TS, or do I have to duplicate every loop member? I was planning to add a few more...

**Comment by elalish** [comment]
@cdata Okay, I'm happy with this now, except that it needs some formatting work to look prettier. Figure I'll wait till the new doc design lands and match that, or else you could give me some pointers. The change to loading.ts is to fix the bug I found where the poster could not be dismissed after changing src.

File: packages/model-viewer/src/features/loading.ts [hunk 1]
```diff
@@ -379,8 +379,8 @@ export const LoadingMixin = <T extends Constructor<ModelViewerElementBase>>(
     get[$shouldAttemptPreload](): boolean {
       const {src} = this;
 
-      return !!src && !CachingGLTFLoader.hasFinishedLoading(src) &&
-          (this.preload || this[$shouldRevealModel]) && this[$isInRenderTree];
```
Comment (cdata) [inline packages/model-viewer/src/features/loading.ts]: Please add a test!

File: packages/model-viewer/src/documentation/tester.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
+/* @license
```
Comment (cdata) [inline packages/model-viewer/src/documentation/tester.ts]: This looks great, thanks for getting it into the build pipeline 💯

File: packages/model-viewer/src/features/loading.ts (hunk 1 above)
Comment (elalish) [reply packages/model-viewer/src/features/loading.ts]: Done!

**Comment by elalish** [comment]
@cdata I added an API to support loading zipped GLTFs: setURLModifier(callback), which just forwards [this](https://threejs.org/docs/index.html#api/en/loaders/managers/LoadingManager.setURLModifier). @donmccurdy I left out a line I found in yours: `loader.setCrossOrigin('anonymous');`. Nothing is broken when serving locally, but I don't know if it'll still work when it's published.

I'm also not sure why it put in a huge package-lock change; maybe it just wasn't included in the docs PR?

File: packages/modelviewer.dev/examples/tester.html [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+      <option value="../shared-assets/environments/spruit_sunrise_1k_HDR.hdr">spruit sunrise HDR</option>
+      <option value="../shared-assets/environments/spruit_sunrise_1k_LDR.jpg">spruit sunrise JPG</option>
+      <option value="../assets/music_hall_01_1k.hdr">music hall</option>
+      <option value="../assets/whipple_creek_regional_park_04_1k.hdr">whipple creek</option>
+      <option value="../shared-assets/environments/aircraft_workshop_01_1k.hdr">aircraft workshop</option>
+      <option value="../assets/pillars_1k.hdr">pillars</option>
+    </select><br/>
+
+    <label for="useSkybox">use as skybox:</label>
+    <input id="useSkybox" type="checkbox"><br/>
```
Comment (cdata) [inline packages/modelviewer.dev/examples/tester.html]: Checking this when `environmentImage` is "default" doesn't do anything. I would suggest either disabling the checkbox in that case (add the `disabled` attribute) or backfilling with the actual IBL we have.

File: packages/modelviewer.dev/examples/tester.html [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+  </style>
+</head>
+<body>
+
+  <div id="container">
+    <div id="instructions">
+    <h1>
+      <a href="https://github.com/GoogleWebComponents/model-viewer" target="_blank">&lt;model-viewer&gt;</a>
+    </h1>
+    Drop a <a href="https://github.com/KhronosGroup/glTF" target="_blank">GLB or zipped GLTF</a> for the model =><br/>
```
Comment (cdata) [inline packages/modelviewer.dev/examples/tester.html]: Sounds like we also support unzipped glTFs, right?

Also: prefer glTF/glB over GLTF/GLB.

File: packages/modelviewer.dev/index.html [hunk 5]
```diff
@@ -500,6 +530,14 @@ <h3 class="grouping-title">Methods</h3>
               this method is invoked, but the model may not update until the next
               render frame.</p>
             </li>
+            <li>
+              <div>setURLModifier(callback)</div>
```
Comment (cdata) [inline packages/modelviewer.dev/index.html]: Let's make this method static (it seems as though it is already effectively static in terms of side-effects, and the last `<model-viewer>` configured will win).

Also, let's take this opportunity to pick a more concise name. I'll suggest: `mapURLs(...)`.

File: packages/modelviewer.dev/examples/tester.html [hunk 6]
```diff
@@ -105,7 +105,7 @@
     <h1>
       <a href="https://github.com/GoogleWebComponents/model-viewer" target="_blank">&lt;model-viewer&gt;</a>
     </h1>
-    Drop a <a href="https://github.com/KhronosGroup/glTF" target="_blank">GLB or zipped GLTF</a> for the model =><br/>
+    Drop a <a href="https://github.com/KhronosGroup/glTF" target="_blank">glB or glTF</a> for the model =><br/>
```
Comment (mrdoob) [inline packages/modelviewer.dev/examples/tester.html]: "Embedded glTF" maybe?
If a user drops a gltf that depends on bin files it won't work.

Comment (donmccurdy) [reply packages/modelviewer.dev/examples/tester.html]: With the updates in this PR, it `.bin` files should work too. See the `simple-dropzone` dependency: https://github.com/GoogleWebComponents/model-viewer/pull/941/files#diff-4adb71782bb4c74dc2a63a7d1f3a533aR22-R23

File: packages/modelviewer.dev/examples/tester.html (hunk 4 above)
Comment (donmccurdy) [reply packages/modelviewer.dev/examples/tester.html]: nit: glTF/GLB. It is... not the most memorable. 😅

File: packages/modelviewer.dev/examples/tester.html (hunk 6 above)
Comment (mrdoob) [reply packages/modelviewer.dev/examples/tester.html]: Oh I see!

File: packages/modelviewer.dev/examples/tester.html [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+  <script type="module"
+      src="./built/dependencies.js">
+  </script>
+  <script nomodule
+      src="./built/dependencies-legacy.js">
+  </script>
+
+  <!-- Loads <model-viewer> only on modern browsers: -->
+      <script type="module"
+      src="../node_modules/@google/model-viewer/dist/model-viewer.js">
```
Comment (cdata) [inline packages/modelviewer.dev/examples/tester.html]: You can remove this `<script>` that pulls in `model-viewer.js`. The code is already being bundled into `tester.js`, so it causes a DOMException when this one loads (attempt to redefine `model-viewer` tag).

File: packages/modelviewer.dev/examples/tester.html [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+      <li class="attribution">
+        <a href="https://hdrihaven.com/hdri/?h=whipple_creek_regional_park_04">whipple_creek_regional_park_04_1k.hdr</a> by <a href="https://hdrihaven.com">HDRI Haven</a>,
+        licensed under <a href="https://hdrihaven.com/p/license.php">CC0</a>.
+      </li>
+    </ul>
+    <div style="margin-top:24px;" class="copyright">©Copyright 2018 Google Inc. Licensed under the Apache License 2.0.</div>
+    <div><a href="https://github.com/GoogleWebComponents/model-viewer">Github</a> ∙ <a href="https://model-viewer.glitch.me/">Glitch</a> ∙ <a href="https://github.com/GoogleWebComponents/model-viewer/issues">Bug report</a></div>
+  </div>
+
+  <script type="module" src="./built/tester.js"></script>
```
Comment (cdata) [inline packages/modelviewer.dev/examples/tester.html]: You will need to build a "legacy" bundle for `tester.js`. Add an entry here: 

https://github.com/GoogleWebComponents/model-viewer/blob/34d8611c58e867272a780f632d719d5b39a7f1d1/packages/modelviewer.dev/scripts/create-legacy-bundles.js#L19

And a UMD bundle config like this one to `rollup.config.js`:

https://github.com/GoogleWebComponents/model-viewer/blob/600bcebf66962059bdd13ba881bea2126eed9b93/packages/modelviewer.dev/rollup.config.js#L39-L48

File: packages/modelviewer.dev/examples/tester.html [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+  </script>
+
+  <!-- Loads <model-viewer> only on modern browsers: -->
+      <script type="module"
+      src="../node_modules/@google/model-viewer/dist/model-viewer.js">
+  </script>
+
+  <!-- Loads <model-viewer> only on old browsers like IE11: -->
+  <script nomodule
+      src="../node_modules/@google/model-viewer/dist/model-viewer-legacy.js">
```
Comment (cdata) [inline packages/modelviewer.dev/examples/tester.html]: Once you have a "legacy" bundle (see comment above), change this to point to `tester-legacy.js`.

File: packages/modelviewer.dev/examples/tester.html [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+    }
+
+    #instructions {
+      width: 320px;
+      margin-top: 10px;
+      margin-left: 30px;
+      margin-bottom: 20px;
+    }
+
+    /* Nokia 8810 */
```
Comment (cdata) [inline packages/modelviewer.dev/examples/tester.html]: lol wat

Comment (mrdoob) [reply packages/modelviewer.dev/examples/tester.html]: I test things everywhere 🤓

**Pull request merged**